delegate-app/Main.hs

```diff
@@ -165,28 +175,31 @@ runDelegateLogicSteps ::
   TQueue (ClientId, FrontendRequest) ->
   -- | the broadcast queue of outgoing messages (write only)
   TChan (ClientResponseScope, DelegateResponse) ->
-  StateT DelegateState (DelegateTracerT IO) void
+  CompositeRunner void
 -- FIXME: we need to abort at some point but this doesn't seem
 -- to be implemented yet so we just go on
 runDelegateLogicSteps
   tick
   eventQueue
   frontendRequestQueue
-  broadcast = forever $ do
-    flushAndTraverseNewElements
-      eventQueue
-      ( delegateEventStep
-          -- All delegateEventStep responses should be brodcasted
-          >=> return . fmap (Broadcast,)
-          >=> putInToClientsQueue
-      )
-    flushAndTraverseNewElements
-      frontendRequestQueue
-      ( delegateFrontendRequestStep
-          >=> putInToClientsQueue
-      )
-    -- FIXME: log queues overload and make tick not-static
-    liftIO $ threadDelay tick
+  broadcast = do
+    (voidResult, _) <-
+      flip runStateT initialState $ forever $ do
```

Comment:
nit: you can use `evalState` here to avoid having to pattern match on void an return it